src/store/modules/stores.js

```diff
@@ -62,11 +62,13 @@ export default {
         }
       }
       dispatch('pickups/setStoreFilter', storeId, {root: true})
+      dispatch('sidenavBoxes/hideGroupSidenav', null, {root: true})
```

Comment:
I didn't really know where to put the two lines
`dispatch('sidenavBoxes/hideGroupSidenav', null, {root: true})` and
`dispatch('sidenavBoxes/showGroupSidenav', null, {root: true})`

Opening and hiding the groupSidenav when you enter/leave a store page works perfectly here though as far as I know.

Reply:
I think this is the best location. In `routes/main.js`, you will find those lines for the store route

```
          beforeEnter: 'stores/selectStore',
          afterLeave: 'stores/clearSelectedStore',
```
Means that those actions will be triggered every time you visit/leave a store page or a child page.